src/main/scala/za/co/absa/faDB/enceladus/DatasetSchema.scala

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2022 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.faDB.enceladus
+
+import slick.dbio.{DBIOAction, NoStream}
+import slick.jdbc.GetResult
+import za.co.absa.faDB.DBFunction.{DBSeqFunction, DBValueFunction}
+import za.co.absa.faDB.enceladus.DatasetSchema.{AddSchema, GetSchema, ListSchemas}
+import za.co.absa.faDB.{DBSchema, DBSession}
+
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
+import slick.jdbc.PostgresProfile.api._
+import slick.sql.SqlStreamingAction
+import za.co.absa.faDB.exceptions.DBFailException
+
+import java.sql.Timestamp
+
+import za.co.absa.faDB.namingConventions.SnakeCaseNaming.Implicits.namingConvention
+
+
+class DatasetSchema(session: DBSession) extends DBSchema(session) {
+  private val db = Database.forConfig(session.connection)
+
+  val addSchema = new AddSchema(this)
+  val getSchema = new GetSchema(this)
+  val listSchemas = new ListSchemas(this)
+
+  def run[R](a: DBIOAction[R, NoStream, Nothing]): Future[R] = {
+    db.run(a)
+  }
+}
+
+object DatasetSchema {
+  case class SchemaInput(schemaName: String,
+                         schemaVersion: Int,
+                         schemaDescription: Option[String],
+                         fields: Option[String],
+                         userName: String)
+  case class Schema(idSchemaVersion: Long,
+                    schemaName: String,
+                    schemaVersion: Int,
+                    schemaDescription: Option[String],
+                    fields: Option[String],
+                    createdBy: String,
+                    createdWhen: Timestamp,
+                    updatedBy: String,
+                    updatedWhen: Timestamp,
+                    lockedBy: Option[String],
+                    lockedWhen: Option[Timestamp],
+                    deletedBy: Option[String],
+                    deletedWhen: Option[Timestamp])
+  case class SchemaHead(schemaName: String, schemaLatestVersion: Int)
+
+  private implicit val SchemaHeadImplicit: GetResult[SchemaHead] = GetResult(r => {SchemaHead(r.<<, r.<<)})
+  private implicit val GetSchemaImplicit: GetResult[Schema] = GetResult(r => {
+    val status: Int = r.<<
+    val statusText: String = r.<<
+    if (status != 200) {
+      throw DBFailException(status, statusText)
+    }
+    Schema(r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<)
```

Comment:
This looks a bit funny. Are there other options?

Reply:
I hate it myself. But I haven't found an easy way to refer to the fields by name in Slick (remember it's ORM focused, the direct SQL is just a side-ability, this is what their docs suggest).
Although it should be possible, it seemed to me. I just didn't want to bother with its implementation for the moment. (A nice implicit enabling it, is another of the #plans 😄 )